src/businessrules/dai/ShopRepository.java

```diff
@@ -3,5 +3,15 @@
 import org.json.JSONObject;
 
 public interface ShopRepository {
+    String createShop(JSONObject data);
+
     boolean updateShop(String shopId, JSONObject shop);
+
+    boolean clearShopMenu(String shopId);
+
+    JSONObject readShop(String shopId);
+
+    JSONObject readShopMenu(String shopId);
```

Comment:
Why is a read shop menu required, can you not just read shop then pull the menu from that?

Reply:
Agreed

Reply:
Do you mean whenever we need to read the Shop Menu we would just call readShop and then parse to get the shop json object within that method? would it be like:
`JSONObject shopData = shopRepository.readShop.getJSONobject("shop")`